### ISSUE TITLE
Jackson Error when doing integration tests with GraphQL

### ISSUE THREAD

**Issue by craigmiller160** [graphql-java-kickstart/graphql-spring-boot#444]
**Describe the bug**
I am trying to write end-to-end integration tests with the latest (7.1.0) version of this library. I am unable to do so because of a Jackson error.

**To Reproduce**

I have the following test GraphQL query file:

```
query {
    clients {
        id
        name
        clientKey
        enabled
        allowAuthCode
        allowClientCredentials
        allowPassword
    }
}
```

I then try to run an integration test that sends it to the endpoint with this code:

```
@ExtendWith(SpringExtension::class)
@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
class ClientQueryIntegrationTest {

    @Autowired
    private lateinit var graphqlRestTemplate: GraphQLTestTemplate

    @Test
    fun test() {
        graphqlRestTemplate.addHeader("Authorization", "Bearer $token")
        val response = graphqlRestTemplate.postForResource("graphql/getAllClients.graphql")
    }

}
```

Then I get the following exception message:

```
com.fasterxml.jackson.databind.exc.MismatchedInputException: Cannot construct instance of `graphql.kickstart.execution.GraphQLRequest` (although at least one Creator exists): no String-argument constructor/factory method to deserialize from String value ('{"query":"query {\n    clients {\n        id\n        name\n        clientKey\n        enabled\n        allowAuthCode\n        allowClientCredentials\n        allowPassword\n    }\n}\n","variables":null}')
 at [Source: (String)""{\"query\":\"query {\\n    clients {\\n        id\\n        name\\n        clientKey\\n        enabled\\n        allowAuthCode\\n        allowClientCredentials\\n        allowPassword\\n    }\\n}\\n\",\"variables\":null}""; line: 1, column: 1]
```

**Expected behavior**

I expect the request to be properly de-serialized by jackson and the test to be able to execute.

**Desktop (please complete the following information):**
 - OS: [e.g. iOS] Ubuntu 18.04
 - Browser [e.g. chrome, safari] N/A
 - Version [e.g. 22] 7.1.0

**Additional context**

Using Spring Boot 2.3.1-RELEASE.

**Comment by craigmiller160** [comment]
Still no response? Does anyone even have a page of documentation to link to? Maybe I'm doing something wrong, I don't know. But any guidance at all would be great. Thanks.

**Comment by craigmiller160** [comment]
Hi everyone. I am writing this to provide the solution to my problem. I finally discovered that it was something in my application that was breaking this. Here is the full explanation, so that anyone who encounters this issue won't have to struggle with it.

**TL;DR** I had overridden the default RestTemplateBuilder bean for special customizations in my application. This resulted in not all of the message converters that are available by default being added in my application. Thus the inability to de-serialize problem that I reported. Moving back onto the default RestTemplateBuilder and applying my changes in another way solved the problem.

**My Bean**
So this is what I had done to override my RestTemplateBuilder. My original goal was adding the interceptor to log requests, but after I did that I got converter errors, so I re-added the two message converters you see there.

```
@Bean
fun restTemplateBuilder(): RestTemplateBuilder {
    return RestTemplateBuilder()
            .interceptors(listOf(RequestResponseLoggingInterceptor()))
            .messageConverters(FormHttpMessageConverter(), MappingJackson2HttpMessageConverter())
}
```

However, by default there are at least 10 message converters added to the RestTemplateBuilder. The lack of the other ones appears to have been the source of my error as to why the GraphQL request wasn't being de-serialized properly.

To get around this, I switched from overriding the RestTemplateBuilder to simply providing a RestTemplateCustomizer Bean to add my interceptor:

```
@Bean
fun formRestTemplateCustomizer(): RestTemplateCustomizer {
    return RestTemplateCustomizer { template ->
        template.interceptors.add(RequestResponseLoggingInterceptor())
    }
}
```

This allowed me to accomplish my goal AND get the integration tests working.

I apologize for opening this issue for my own mistake. Closing it now.
